`aws-go-eks` is broken (after v4 update?)

### What happened?

I tried to run the `aws-go-eks` example by running `pulumi up`, but I got an error. I expect it might happened due to the recent update of sdk to v4:
```
	"github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes"
	appsv1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/apps/v1"
	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/core/v1"
	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/meta/v1"
```
When I downgrade to v3 and go `go get` I get a step farther with `pulumi up`, but that eventually also fails.

### Example with v4

```
$ pulumi up
Previewing update (dev)

View in Browser (Ctrl+O): https://app.pulumi.com/zdrapela/aws-go-eks/dev/previews/32e432a3-d4db-4174-9245-8487e3033aca

     Type                 Name            Plan     Info
     pulumi:pulumi:Stack  aws-go-eks-dev           1 error; 3 messages

Diagnostics:
  pulumi:pulumi:Stack (aws-go-eks-dev):
    # aws-go-eks
    ./main.go:173:35: namespace.Metadata.Elem undefined (type "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/meta/v1".ObjectMetaOutput has no field or method Elem)
    ./main.go:201:35: namespace.Metadata.Elem undefined (type "github.com/pulumi/pulumi-kubernetes/sdk/v4/go/kubernetes/meta/v1".ObjectMetaOutput has no field or method Elem)

    error: error in compiling Go: unable to run `go build`: exit status 1
```

### Example with v3 on `us-east-1`
_I got a failure with `us-east-1`. After setting `eu-north-1` as a default, I was able to successfully create the cluster._
```
$ pulumi up
Previewing update (dev)

View in Browser (Ctrl+O): https://app.pulumi.com/zdrapela/aws-go-eks/dev/previews/5cae7382-990a-4336-ae90-a7a0812dc941

     Type                              Name                Plan       
 +   pulumi:pulumi:Stack               aws-go-eks-dev      create     
 +   ├─ aws:iam:Role                   eks-iam-eksRole     create     
 +   ├─ aws:iam:Role                   nodegroup-iam-role  create     
 +   ├─ aws:iam:RolePolicyAttachment   ngpa-0              create     
 +   ├─ aws:iam:RolePolicyAttachment   ngpa-1              create     
 +   ├─ aws:iam:RolePolicyAttachment   rpa-1               create     
 +   ├─ aws:ec2:SecurityGroup          cluster-sg          create     
 +   ├─ aws:iam:RolePolicyAttachment   ngpa-2              create     
 +   ├─ aws:iam:RolePolicyAttachment   rpa-0               create     
 +   ├─ aws:eks:Cluster                eks-cluster         create     
 +   ├─ aws:eks:NodeGroup              node-group-2        create     
 +   ├─ pulumi:providers:kubernetes    k8sprovider         create     
 +   ├─ kubernetes:core/v1:Namespace   app-ns              create     
 +   ├─ kubernetes:apps/v1:Deployment  app-dep             create     
 +   └─ kubernetes:core/v1:Service     app-service         create     

Outputs:
    kubeconfig: output<string>
    url       : output<string>

Resources:
    + 15 to create

Do you want to perform this update? yes
Updating (dev)

View in Browser (Ctrl+O): https://app.pulumi.com/zdrapela/aws-go-eks/dev/updates/24

     Type                             Name                Status                  Info
 +   pulumi:pulumi:Stack              aws-go-eks-dev      **creating failed**     1 error
 +   ├─ aws:iam:Role                  nodegroup-iam-role  created (1s)            
 +   ├─ aws:iam:Role                  eks-iam-eksRole     created (2s)            
 +   ├─ aws:ec2:SecurityGroup         cluster-sg          created (5s)            
 +   ├─ aws:iam:RolePolicyAttachment  ngpa-1              created (0.99s)         
 +   ├─ aws:iam:RolePolicyAttachment  ngpa-2              created (1s)            
 +   ├─ aws:iam:RolePolicyAttachment  ngpa-0              created (1s)            
 +   ├─ aws:iam:RolePolicyAttachment  rpa-0               created (1s)            
 +   ├─ aws:iam:RolePolicyAttachment  rpa-1               created (1s)            
 +   └─ aws:eks:Cluster               eks-cluster         **creating failed**     2 errors

Diagnostics:
  aws:eks:Cluster (eks-cluster):
    error:   sdk-v2/provider2.go:515: sdk.helper_schema: creating EKS Cluster (eks-cluster-82ad2bb): operation error EKS: CreateCluster, https response error StatusCode: 400, RequestID: 8a1fd60e-64a8-40ca-a829-6f3a5e65171f, UnsupportedAvailabilityZoneException: Cannot create cluster 'eks-cluster-82ad2bb' because EKS does not support creating control plane instances in us-east-1e, the targeted availability zone. Retry cluster creation using control plane subnets that span at least two of these availability zones: us-east-1a, us-east-1b, us-east-1c, us-east-1d, us-east-1f. Note, post cluster creation, you can run worker nodes in separate subnets/availability zones from control plane subnets/availability zones passed during cluster creation: provider=aws@6.67.0
    error: 1 error occurred:
    	* creating EKS Cluster (eks-cluster-82ad2bb): operation error EKS: CreateCluster, https response error StatusCode: 400, RequestID: 8a1fd60e-64a8-40ca-a829-6f3a5e65171f, UnsupportedAvailabilityZoneException: Cannot create cluster 'eks-cluster-82ad2bb' because EKS does not support creating control plane instances in us-east-1e, the targeted availability zone. Retry cluster creation using control plane subnets that span at least two of these availability zones: us-east-1a, us-east-1b, us-east-1c, us-east-1d, us-east-1f. Note, post cluster creation, you can run worker nodes in separate subnets/availability zones from control plane subnets/availability zones passed during cluster creation

  pulumi:pulumi:Stack (aws-go-eks-dev):
    error: update failed

Resources:
    + 9 created

Duration: 16s

```

### Output of `pulumi about`

```
CLI          
Version      3.147.0
Go Version   go1.23.5
Go Compiler  gc

Plugins
KIND      NAME        VERSION
resource  aws         6.67.0
language  go          3.147.0
resource  kubernetes  4.21.1
resource  kubernetes  3.30.2

Host     
OS       darwin
Version  15.2
Arch     arm64

This project is written in go: executable='/usr/local/go/bin/go' version='go version go1.23.5 darwin/arm64'

Current Stack: zdrapela/aws-go-eks/dev

TYPE                  URN
pulumi:pulumi:Stack   urn:pulumi:dev::aws-go-eks::pulumi:pulumi:Stack::aws-go-eks-dev
pulumi:providers:aws  urn:pulumi:dev::aws-go-eks::pulumi:providers:aws::default_6_67_0


Found no pending operations associated with dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/zdrapela
User           zdrapela
Organizations  zdrapela
Token type     personal

Dependencies:
NAME                                        VERSION
github.com/pulumi/pulumi-aws/sdk/v6         v6.67.0
github.com/pulumi/pulumi-kubernetes/sdk/v3  v3.30.2
github.com/pulumi/pulumi-kubernetes/sdk/v4  v4.21.1
github.com/pulumi/pulumi/sdk/v3             v3.147.0

Pulumi locates its logs in /var/folders/7v/nnjpghy11y1gl47f9jjckjnh0000gn/T/ by default
```

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/aws-go-eks/README.md
+++ b/aws-go-eks/README.md
@@ -1,7 +1,8 @@
+# AWS Golang EKS Cluster
+
 [![Deploy this example with Pulumi](https://www.pulumi.com/images/deploy-with-pulumi/dark.svg)](https://app.pulumi.com/new?template=https://github.com/pulumi/examples/blob/master/aws-go-eks/README.md#gh-light-mode-only)
 [![Deploy this example with Pulumi](https://get.pulumi.com/new/button-light.svg)](https://app.pulumi.com/new?template=https://github.com/pulumi/examples/blob/master/aws-go-eks/README.md#gh-dark-mode-only)
 
-# AWS Golang EKS Cluster
 This example creates an AWS EKS Cluster and deploys a sample container application to it
 
 ## Deploying the App
@@ -14,7 +15,7 @@ This example creates an AWS EKS Cluster and deploys a sample container applicati
 2. [Install Go](https://golang.org/doc/install)
 3. [Configure AWS Credentials](https://www.pulumi.com/docs/intro/cloud-providers/aws/setup/)
 4. [Install `aws-iam-authenticator`](https://docs.aws.amazon.com/eks/latest/userguide/install-aws-iam-authenticator.html)
-4. [Install `kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
+5. [Install `kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/)
 
 ### Steps
 
@@ -23,44 +24,43 @@ After cloning this repo, run these commands from the working directory:
 1. Create a new stack, which is an isolated deployment target for this example:
 
     ```bash
-    $ pulumi stack init dev
+    pulumi stack init dev
     ```
 
 2. Set your desired AWS region:
 
     ```bash
-    $ pulumi config set aws:region us-east-1 # any valid AWS region will work
+    pulumi config set aws:region us-east-1 # any valid AWS region will work
     ```
 
-4. Execute the Pulumi program to create our EKS Cluster:
+3. Execute the Pulumi program to create our EKS Cluster:
 
-	```bash
-	pulumi up
-	```
+    ```bash
+    pulumi up
+    ```
 
-5. After 10-15 minutes, your cluster will be ready, and the kubeconfig JSON you'll use to connect to the cluster will
+4. After 10-15 minutes, your cluster will be ready, and the kubeconfig JSON you'll use to connect to the cluster will
    be available as an output. You can save this kubeconfig to a file like so:
 
     ```bash
-    $ pulumi stack output kubeconfig --show-secrets >kubeconfig.json
+    pulumi stack output kubeconfig --show-secrets >kubeconfig.json
     ```
 
     Once you have this file in hand, you can interact with your new cluster as usual via `kubectl`:
 
     ```bash
-    $ KUBECONFIG=./kubeconfig.json kubectl get nodes
+    KUBECONFIG=./kubeconfig.json kubectl get nodes
     ```
 
-6. Ensure that the application is running as expected:
+5. Ensure that the application is running as expected:
 
     ```bash
-   $ curl $(pulumi stack output url)
+   curl $(pulumi stack output url)
    ```
 
+6. Afterwards, destroy your stack and remove it:
 
-7. Afterwards, destroy your stack and remove it:
-
-	```bash
-	pulumi destroy --yes
-	pulumi stack rm --yes
-	```
+ ```bash
+ pulumi destroy --yes
+ pulumi stack rm --yes
+ ```

--- a/aws-go-eks/go.mod
+++ b/aws-go-eks/go.mod
@@ -1,8 +1,6 @@
 module aws-go-eks
 
-go 1.22
-
-toolchain go1.24.0
+go 1.24
 
 require (
 	github.com/pulumi/pulumi-aws/sdk/v6 v6.69.0

--- a/aws-go-eks/main.go
+++ b/aws-go-eks/main.go
@@ -145,12 +145,11 @@ func main() {
 			return err
 		}
 
-		ctx.Export("kubeconfig", generateKubeconfig(eksCluster.Endpoint,
-			eksCluster.CertificateAuthority.Data().Elem(), eksCluster.Name))
+		kubeconfig := generateKubeconfig(eksCluster.Endpoint, eksCluster.CertificateAuthority.Data().Elem(), eksCluster.Name)
 
+		ctx.Export("kubeconfig", kubeconfig)
 		k8sProvider, err := kubernetes.NewProvider(ctx, "k8sprovider", &kubernetes.ProviderArgs{
-			Kubeconfig: generateKubeconfig(eksCluster.Endpoint,
-				eksCluster.CertificateAuthority.Data().Elem(), eksCluster.Name),
+			Kubeconfig: kubeconfig,
 		}, pulumi.DependsOn([]pulumi.Resource{nodeGroup}))
 		if err != nil {
 			return err
@@ -170,7 +169,7 @@ func main() {
 		}
 		_, err = appsv1.NewDeployment(ctx, "app-dep", &appsv1.DeploymentArgs{
 			Metadata: &metav1.ObjectMetaArgs{
-				Namespace: namespace.Metadata.Elem().Name(),
+				Namespace: namespace.Metadata.Name(),
 			},
 			Spec: appsv1.DeploymentSpecArgs{
 				Selector: &metav1.LabelSelectorArgs{
@@ -198,7 +197,7 @@ func main() {
 
 		service, err := corev1.NewService(ctx, "app-service", &corev1.ServiceArgs{
 			Metadata: &metav1.ObjectMetaArgs{
-				Namespace: namespace.Metadata.Elem().Name(),
+				Namespace: namespace.Metadata.Name(),
 				Labels:    appLabels,
 			},
 			Spec: &corev1.ServiceSpecArgs{
@@ -253,10 +252,11 @@ func generateKubeconfig(clusterEndpoint pulumi.StringOutput, certData pulumi.Str
             "user": {
                 "exec": {
                     "apiVersion": "client.authentication.k8s.io/v1beta1",
-                    "command": "aws-iam-authenticator",
+                    "command": "aws",
                     "args": [
-                        "token",
-                        "-i",
+                        "eks",
+                        "get-token",
+                        "--cluster-name",
                         "%s",
                     ],
                 },

